python/rmm/_lib/device_buffer.pyx

```diff
@@ -452,13 +438,6 @@ cpdef void copy_device_to_ptr(uintptr_t d_src,
     d_dst : pointer to data on device to write into
     stream : CUDA stream to use for copying, default the default stream
 
-    Note
-    ----
-
-    If ``stream`` is the default stream, it is synchronized after the copy.
-    However if a non-default ``stream`` is provided, this function is fully
-    asynchronous.
```

Comment:
I think we need to synchronize if it's the default stream here as well. If `d_src` is a host pointer it can go out of scope and be freed before the copy finishes. There used to be a synchronization in `_copy_async` that we've rightfully removed, so we need to add it back into here.

Reply:
Apologies, misread the API because of the docstring, we don't need to synchronize here.

Reply:
Done.